openlibrary/book_providers.py

```diff
@@ -21,6 +29,96 @@ class EbookAccess(OrderedEnum):
     def to_solr_str(self):
         return self.name.lower()
 
+    @staticmethod
+    def from_provider_literal(literal: ProviderAccessLiteral) -> 'EbookAccess':
+        if literal == 'sample':
+            # We need to update solr to handle these! Requires full reindex
+            return EbookAccess.PRINTDISABLED
+        elif literal == 'buy':
+            return EbookAccess.NO_EBOOK
+        elif literal == 'open-access':
+            return EbookAccess.PUBLIC
+        elif literal == 'borrow':
+            return EbookAccess.BORROWABLE
+        elif literal == 'subscribe':
+            return EbookAccess.NO_EBOOK
+        else:
+            raise ValueError(f'Unknown access literal: {literal}')
```

Comment:
```suggestion
        # We need to update solr to handle samples! Requires full reindex
        try:
            return {
                'sample': EbookAccess.PRINTDISABLED,
                'buy': EbookAccess.NO_EBOOK,
                'open-access': EbookAccess.PUBLIC,
                'borrow': EbookAccess.BORROWABLE,
                'subscribe': EbookAccess.NO_EBOOK,
            }[literal]
        except KeyError:
            raise ValueError(f'Unknown access literal: {literal}')
```

Reply:
Scott and I chatted and we think the current implementation is a little easier to read. Thank you for the suggestion though!